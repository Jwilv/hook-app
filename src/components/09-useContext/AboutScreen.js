import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {
    const {user,setUser} = useContext(UserContext)
    const handlelimpiar = ()=>{
        setUser({})
    }
    return (
        <div>
            <h1>AboutScreen</h1>
            <hr />
            <h2>{JSON.stringify(user,null,4)}</h2>
            <button
            className='btn btn-primary'
            onClick={ ()=>{
                setUser({...user,nombrePadre:'pedro paramo'})
            }}>
                Agregar al padre
            </button>
            <button
            className='btn btn-primary'
            onClick={ handlelimpiar }>
                limpiar
            </button>
        </div>
    )
}
