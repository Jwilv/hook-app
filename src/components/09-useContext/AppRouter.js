import React from 'react'
import {
    Routes,
    Route,
} from 'react-router-dom'
import { AboutScreen } from './AboutScreen'
import { HomeScreen } from './HomeScreen'
import { LoginScreen } from './LoginScreen'
import { NavBar } from './NavBar'


export const AppRouter = () => {
    return (
        <>
            <NavBar />
            <Routes>
                <Route path='/' element={<HomeScreen />}/>
                <Route path='/about' element={<AboutScreen />} />
                <Route path='/login' element={<LoginScreen />} />
                <Route path='*' element={<h1>404 alla la ponen en 4</h1>} />
                
            </Routes>
        </>
    )
}
