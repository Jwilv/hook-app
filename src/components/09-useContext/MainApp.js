import React from 'react'
// import {
//     Routes,
//     Route,
// } from 'react-router-dom'
// import { AboutScreen } from './AboutScreen'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {
    const user = {
        name :"juan",
        id : 1234,
    }
    return (
        <UserContext.Provider value={user}>
            <AppRouter/>
        </UserContext.Provider>
    )
}
