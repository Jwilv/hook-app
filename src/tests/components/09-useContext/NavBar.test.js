import React from 'react'
import { shallow } from 'enzyme'
import { NavBar } from '../../../components/09-useContext/NavBar';

describe('pruebas en <NavBar />', () => { 

    const wrapper = shallow( <NavBar />)

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
 })