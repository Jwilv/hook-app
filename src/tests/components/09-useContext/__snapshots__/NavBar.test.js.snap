// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <NavBar /> debe de mostrarse correctamente 1`] = `
<nav>
  <ul>
    <li>
      <Link
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <Link
        to="/about"
      >
        About
      </Link>
    </li>
    <li>
      <Link
        to="/login"
      >
        Login
      </Link>
    </li>
  </ul>
</nav>
`;
